CLI checks for AWS credentials

TommyJackson85 opened the pull request:
## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
Closes #713

## Your solution
<!--- Please describe your solution, have you encountered any issues along the way? -->
Checking for aws sts credentials from a new plugin.

## How Has This Been Tested?
Regularly ran ```npx webiny deploy api --env=local``` from sample-project folder
Console logged succesfull passes originally and error messages originally.

Will continue to look through aws documentation to double check my methods and what I need to do next.

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if relevant):

Pavel910 commented on packages/cli-plugin-deploy-components/aws/index.js:
```diff
@@ -0,0 +1,27 @@
+
+module.exports = {
+    type: "hook-before-deploy",
+    name: "cli-command-aws",
```
Change the name of the plugin to `hook-before-deploy-aws-credentials`

Pavel910 commented on packages/cli-plugin-deploy-components/aws/index.js:
```diff
@@ -0,0 +1,27 @@
+    async hook() {
+        // check AWS credentials
+        const sts = new STS();
+        const stsResults = await sts
+            .getCallerIdentity({})
+            .promise()
+            .then((result, err) => {
+                if (err) {
+                    console.err(`The following error occured: ${err}.`);
+                    console.log(`Please to refer to the following link for setting up credentials:
+                    Please to refer to the following link for setting up aws credentials:
+                    https://docs.webiny.com/docs/guides/aws-credentials/
+                    `);
+                } else {
+                    console.log("Your AWS data is set!");
+                    console.log(result);
+                }
+            });
+    }
```
This part can be written a lot simpler and cleaner:

```js
try {
    await sts.getCallerIdentity({}).promise();
} catch(err) {
    console.log(`Looks like your AWS credentials are not configured correctly!`);
    // .... link to docs ....
}
```

TommyJackson85 replied:
@Pavel910 sorry! I misinterpreted one of your posts from the issue post. I will bring back try and catch so.

TommyJackson85 replied:
@Pavel910 Do you want a message to appear if aws credentials pass?

TommyJackson85 commented:
@Pavel910 Do you want a message to appear if aws credentials pass?

Pavel910 commented:
@TommyJackson85 no need, we don't want to spam users if there's no actual use for that message. If credentials are fine - just move on with the process. If not - that's where we want user's attention.

Pavel910 commented on packages/cli-plugin-deploy-components/aws/index.js:
```diff
@@ -0,0 +1,19 @@
+        } catch (err) {
+            console.log("Looks like your AWS credentials are not configured correctly!");
+            console.log(
+                "Please to refer to the following link for setting them up: https://docs.webiny.com/docs/guides/aws-credentials/"
```
This sentence looks off to me: `Please to refer to the following...` ?

TommyJackson85 commented:
@Pavel910 any suggestions on the kind of message we should display? I updated it, as I think you were right, it was too complex. It was a little bit long too. Let me know what you think of my changes.

Pavel910 commented:
@TommyJackson85 it wasn't too long, just had an extra `to` which made it totally weird.

I think we can do something like: 
```
To learn how to configure your AWS credentials, visit https://docs.webiny.com/docs/guides/aws-credentials.
```

TommyJackson85 commented:
@Pavel910 So I updated the console message again, and I removed the comment from the hook plugin and I removed the aws.js file as it ended up not being used and was empty.
Was ready to suggest a merge but now I have build package -> commodo error for some reason :

<img width="1037" alt="Screenshot 2020-07-10 at 19 28 51" src="https://user-images.githubusercontent.com/22279028/87186626-bb106500-c2e3-11ea-9d48-04914b9de0e3.png">

I dont see why I would get this as it didn't exist before in previous checks. Any suggestions?

doitadrian commented:
Try removing this: `typings/shared/index.d.ts:30`.

TommyJackson85 commented:
> Try removing this: `typings/shared/index.d.ts:30`, we added this a couple of days ago. There was a TS error in a 3rd party `@types` package, which was preventing us from doing a release.
> 
> Maybe they've fixed it already.
> 
> UPDATE: it looks like the issue got closed / resolved ([DefinitelyTyped/DefinitelyTyped#45927](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/45927)).

Well for further context, here is the file you suggested, from my repository:

<img width="642" alt="Screenshot 2020-07-10 at 21 08 43" src="https://user-images.githubusercontent.com/22279028/87196411-e7cb7900-c2f1-11ea-9569-cd391456c747.png">

I did also notice there were some file setup changes , like the removal of sample-project folder, bringing apps folder to the root folder. Could this be related to it?

TommyJackson85 commented:
I pulled the master branch again just to keep it updated. But I had to fix an error in a handler.js file to do with no curly braces, in order to commit my changes. Lets hope this passes!

TommyJackson85 commented:
same error again

TommyJackson85 commented:
> Try removing this: `typings/shared/index.d.ts:30`, we added this a couple of days ago. There was a TS error in a 3rd party `@types` package, which was preventing us from doing a release.
> 
> Maybe they've fixed it already.
> 
> UPDATE: it looks like the issue got closed / resolved ([DefinitelyTyped/DefinitelyTyped#45927](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/45927)).

@doitadrian Is the file ok here? This is the most updated version:

<img width="757" alt="Screenshot 2020-07-11 at 01 18 36" src="https://user-images.githubusercontent.com/22279028/87212049-7ea92d00-c314-11ea-89bb-bdea15ca8711.png">

doitadrian commented:
Sorry, I'm kinda confused. Did you try removing the `typings/shared/index.d.ts:30`?

Thanks!

TommyJackson85 commented:
> Sorry, I'm kinda confused. Did you try removing the `typings/shared/index.d.ts:30`?
> 
> Thanks!
> 
> UPDATE: from what I can see, this wasn't removed. Please remove it and try again.

You said the issue was resolved and I pulled the most recent master branch. No worries though, I removed the interface Module at line 30, as you were referring to that. If it doesn't work, I will try and remove the namespace NodeJS object.

Pavel910 commented:
@TommyJackson85 Adrian was referring to this issue https://github.com/DefinitelyTyped/DefinitelyTyped/issues/45927.
It's the reason why that TS error is happening in the first place.

Since you pulled the changes from Webiny `master` today and installed the dependencies *today*, it's possible that the underlying issue linked above was resolved and now our hotfix is causing the clash with the original code that had to be fixed.

I'll take a look at the code and merge it later, thanks!

TommyJackson85 commented:
@Pavel910 no problem. It would be good to double check my file changes so they don't cause errors but they're passing all checks now.

doitadrian commented on typings/shared/index.d.ts:
```diff
@@ -27,7 +27,4 @@ declare module "*.svg" {
 
 // Hot fix - check https://github.com/DefinitelyTyped/DefinitelyTyped/issues/45927 for a fix.
```
Please delete this while section, including the comment and the `declare namespace ...` statement.

TommyJackson85 commented:
> Minor change, and if you ask me, this looks good.

@doitadrian Will get around to this later. I apologise for the delay!

TommyJackson85 commented:
@doitadrian deleted what you suggested, just double check it to make sure.
